### PR TITLE
Week 2: TDD Module

### PR DESCRIPTION
**Pull request by karngyan**
# Description

- Solved the TDD Module scheduling problem
- Added a Maven Github Action to check if tests pass for the tdd project. Check Status [Here](https://github.com/karngyan/software-product-sprint/pull/8/checks?check_run_id=885919406)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# This workflow will build a the Week 4 TDD Project and runs all Tests
```
Comment (arijit-google-sps) [inline .github/workflows/maven.yml]: Nice!

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+on:
+  push:
+    branches: [ dev-week4-tdd ]
+  pull_request:
+    branches: [ master, dev ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (arijit-google-sps) [inline .github/workflows/maven.yml:14]: only ubuntu?

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,92 @@
...
+   *
+   * @param events Collection of Events
+   * @param request MeetingRequest
+   * @return Collection of potential Time Ranges for Meeting
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // initialize array with true
+    isAvailableAtMinute = initializeIsAvailableAtMinuteArray(0);
```
Comment (arijit-google-sps) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: isMinuteAvailableForAllAttendees can be a better name?

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -14,10 +14,92 @@
...
+
+  /**
+   * > 0 : false
+   * = 0 : true
+   */
+  private List<Integer> isAvailableAtMinute;
+
+  /**
+   * Worst Case
+   * Time Complexity: O(3 * 24 * 60 + events.size() * attendees.size())
```
Comment (arijit-google-sps) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I think this is anyways O(1), so should we remove this?

**Review by arijit-google-sps — APPROVED** [review]
Please take a look at the comments before merging.

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (karngyan) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Sure. Shall update the same.

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 4 above)
Comment (karngyan) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Yep. Done.

File: .github/workflows/maven.yml (hunk 2 above)
Comment (karngyan) [reply .github/workflows/maven.yml:14]: This is the machine on which the tests are run by GitHub.  Should I run on more platforms?

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (karngyan) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Done.

File: .github/workflows/maven.yml (hunk 2 above)
Comment (arijit-google-sps) [reply .github/workflows/maven.yml:14]: No need, this should be good.

**Pull request merged**